packages/devtools/lib/src/timeline/cpu_profile_model.dart

```diff
@@ -142,56 +148,49 @@ class CpuStackFrame {
 
   final String url;
 
-  final List<CpuStackFrame> children = [];
-
-  CpuStackFrame parent;
-
-  /// Index in [parent.children].
-  int index = -1;
+  // Time data for stack frame's enclosing CPU profile.
+  final TimeRange profileTime;
 
   /// How many cpu samples for which this frame is a leaf.
   int exclusiveSampleCount = 0;
 
-  /// Depth of this CpuStackFrame tree, including [this].
-  ///
-  /// We assume that CpuStackFrame nodes are not modified after the first time
-  /// [depth] is accessed. We would need to clear the cache if this was
-  /// supported.
-  int get depth {
-    if (_depth != 0) {
-      return _depth;
-    }
-    for (CpuStackFrame child in children) {
-      _depth = max(_depth, child.depth);
-    }
-    return _depth = _depth + 1;
-  }
-
-  int _depth = 0;
-
   int get inclusiveSampleCount =>
       _inclusiveSampleCount ?? _calculateInclusiveSampleCount();
 
   /// How many cpu samples this frame is included in.
   int _inclusiveSampleCount;
 
-  double get cpuConsumptionRatio => _cpuConsumptionRatio ??=
-      inclusiveSampleCount / getRoot().inclusiveSampleCount;
+  double get totalTimeRatio => _totalTimeRatio ??=
+      inclusiveSampleCount / (root as CpuStackFrame).inclusiveSampleCount;
 
-  double _cpuConsumptionRatio;
+  double _totalTimeRatio;
 
-  void addChild(CpuStackFrame child) {
-    children.add(child);
-    child.parent = this;
-    child.index = children.length - 1;
-  }
+  Duration get totalTime => _totalTime ??= Duration(
+      microseconds:
+          (totalTimeRatio * profileTime.duration.inMicroseconds).round());
+
+  Duration _totalTime;
+
+  double get selfTimeRatio => _selfTimeRatio ??=
+      exclusiveSampleCount / (root as CpuStackFrame).inclusiveSampleCount;
+
+  double _selfTimeRatio;
+
+  Duration get selfTime => _selfTime ??= Duration(
+      microseconds:
+          (selfTimeRatio * profileTime.duration.inMicroseconds).round());
+
+  Duration _selfTime;
 
-  CpuStackFrame getRoot() {
-    CpuStackFrame root = this;
-    while (root.parent != null) {
-      root = root.parent;
+  /// Returns a simplified version of the url with [path/to/flutter] removed.
+  String get simplifiedUrl {
```

Comment:
move this helper to a core location. could even be its own file of utils manipulating urls if we don't have one already. 
We should probably also be supporting flutter_web here for future compatibility.

Reply:
done and done.